docs/source/tutorial_metrics.md

```diff
@@ -0,0 +1,372 @@
+        if 'hello' in model_text:
+            # count 1 / 1 messages having "hello"
+            self.metrics.add('hello_avg', AverageMetric(1, 1))
+        else:
+            # count 0 / 1 messages having "hello"
+            self.metrics.add('hello_avg', AverageMetric(0, 1))
+
+if __name__ == '__main__':
+    from parlai.scripts.eval_model import EvalModel
+
+    EvalModel.main(
+       task='hello_daily',
+       model_file='zoo:blender/blender_90M/model',
+       batchsize=32,
+    )
+```
+
+If we run the script, we will have a new metric in our output:
+
+```
+18:07:30 | Finished evaluating tasks ['hello_daily'] using datatype valid
+    accuracy  bleu-4  ctpb  ctps  exps  exs    f1  gpu_mem  hello_avg  loss  ltpb  ltps   ppl  token_acc  tpb   tps
+           0 .002035  2172   230 3.351 8069 .1346   .05211      .1228 2.979 495.9 52.52 19.67      .4133 2668 282.6
+```
+
+__What is AverageMetric?__
+
+Wait, what is this
+[AverageMetric](parlai.core.metrics.AverageMetric)? All metrics
+you want to create in ParlAI should be a
+[Metric](Metric) object. Metric objects
+define a way of instantiating the metric, a way of combining it with a
+like-metric, and a way of rendering it as a single float value. For an
+AverageMetric, this means we need to define a numerator and a denominator; the
+combination of AverageMetrics adds their numerators and denominators
+separately. As we do this across all examples, the numerator will be the number
+of examples with "hello" in it, and the denominator will be the total number of
+examples. When we go to print the metric, the division will be computed at the
+last second.
+
+If you're used to writing machine learning code in one-off scripts, you may ask
+why do I need to use this metric? Can't I just count and divide myself? While
+you can do this, your code could not be run in [_distributed
+mode_](tutorial_fast). If we only returned a single float, we would not be able
+to know if some distributed workers received more or fewer examples than
+others. However, when we explicitly store the numerator and denominator, we can
+combine and reduce the across multiple nodes, enabling us to train on hundreds
+of GPUs, while still ensuring correctness in all our metrics.
+
+In addition to AverageMetric, there is also
+[SumMetric](parlai.core.metrics.SumMetric), which keeps a running
+sum. SumMetric and AverageMetric are the most common ways to construct custom
+metrics, but others exist as well. For a full list (and views into advanced
+cases), please see the [metrics API documentation](metrics_api).
+
+## Agent (model) level metrics
+
+In the above example, we worked on a metric defined by a Teacher. However,
+sometimes our models will have special metrics that only they want to compute,
+which we call an Agent-level metric. Perplexity is one example.
+
+To compute model-level metrics, we can define either a global metric, or a
+local metric. Global metrics can be computed anywhere, and are easy to use,
```

Comment:
I think making 'global' and 'local' bold (or any other emphasis) might make this read better. And prepare reader for what comes next.